Automattic/wp-calypso
# Add "Connect" and "Transfer" options to the "Use a domain I own" flow.

#### Changes proposed in this Pull Request

This PR adds in the second page of the "Use a domain I own" flow. So, once the user has selected a domain they want to use, we need to do an availability check on this domain to see if we're able to transfer and/or connect the domain to the user's site.

If the domain is transferable, the option to transfer the domain will be at the top of the second page and the "Select" button for this option will be the primary button.

If the domain is not transferable, then the option to connect will be at the top of the page and the "Select" button for this option will be at the top of the page.

If the domain is not able to be transferred or connected, then an error message will be shown on the first page.

The "Select" buttons on the second page are not yet active. These will be made active in a future PR. The main focus of this PR is to properly check the availability of the domain and present the correct error message or transfer/connect options.

Examples of the primary scenarios are shown in the screenshots below:

Domain is able to be connected or transferred. Site has a paid plan and a domain credit:
<img width="1076" alt="Screen Shot 2021-08-16 at 2 39 45 PM" src="https://user-images.githubusercontent.com/1379730/129762948-e107e31e-4910-44d8-862a-8f8a93c9b28f.png">

Domain is able to be connected or transferred. Site with a paid plan and no credit:
<img width="1076" alt="Screen Shot 2021-08-16 at 2 46 09 PM" src="https://user-images.githubusercontent.com/1379730/129763106-86c8fbbe-3ce4-41f4-8d08-e83a717a3c39.png">

Domain is on sale and able to be connected or transferred. Site with a paid plan and no credit:
<img width="1076" alt="Screen Shot 2021-08-16 at 2 30 46 PM" src="https://user-images.githubusercontent.com/1379730/129763341-7d1b78aa-c8db-462a-97ac-988267ebe7c7.png">

Domain is able to be connected or transferred. Site has no paid plan:
<img width="1076" alt="Screen Shot 2021-08-16 at 2 47 35 PM" src="https://user-images.githubusercontent.com/1379730/129763487-8534e070-cb6d-44b1-99b0-2e3c7ce0373a.png">

Domain is on sale and able to be connected or transferred. Site has no paid plan:
<img width="1076" alt="Screen Shot 2021-08-16 at 2 41 03 PM" src="https://user-images.githubusercontent.com/1379730/129763556-a3c6793e-66d4-486f-91f6-678fc825df34.png">

Domain is able to be connected, but not transferred. Site has a paid plan:
<img width="1076" alt="Screen Shot 2021-08-17 at 12 29 30 PM" src="https://user-images.githubusercontent.com/1379730/129764860-59b3586a-9a29-4f36-a848-28f577d67704.png">

Domain is able to be connected, but not transferred. Site has a no paid plan:
<img width="1076" alt="Screen Shot 2021-08-16 at 2 56 30 PM" src="https://user-images.githubusercontent.com/1379730/129765476-26a49162-fb03-4d8b-936b-0e17a840e18d.png">

Domain is able to be transferred, but not connected. Site has a paid plan and no credit:
<img width="1076" alt="Screen Shot 2021-08-16 at 5 30 31 PM" src="https://user-images.githubusercontent.com/1379730/129766514-411c7257-9c9a-43e2-95b1-974c29619bae.png">

Requested domain is not registered:
<img width="1076" alt="Screen Shot 2021-08-16 at 2 41 58 PM" src="https://user-images.githubusercontent.com/1379730/129766644-56a5b7d4-9b95-47cd-8d03-1d89d704b223.png">

Search query is not a valid FQDN:
<img width="1076" alt="Screen Shot 2021-08-16 at 2 42 07 PM" src="https://user-images.githubusercontent.com/1379730/129766741-fdb3d8a9-829d-4606-99c3-9d7c2826d625.png">

No domain entered:
<img width="1076" alt="Screen Shot 2021-08-16 at 2 42 14 PM" src="https://user-images.githubusercontent.com/1379730/129766827-cec74cbf-b937-45b9-9bf4-08ba19a273db.png">


#### Testing instructions

Go to the path: `/domains/add/use-my-domain/[site slug]`

Try testing the page with each of the scenarios above.

Also try with as many of the various combinations of availability error conditions as you can.

When adding the query string `?initialQuery=[domain name to connect]`, the page should automatically check the availability and load the second page.
